### PR TITLE
Prefix public C functions with ts_

### PR DESCRIPTION
**Pull request by JLockerman**
We've decided to adopt the `ts_` prefix on all exported C functions in
order to avoid having symbol conflicts with future postgres functions.
We've already started using this prefix on new functions and this commit
adds the prefix to to the old functions publicly exported in the timescale binary.
This change ended touching more of the test code than I expected.

Setting @erimatnor as a reviewer since he requested the change, the second reviewer can be anyone.

### CODE REVIEW

**Comment by JLockerman** [comment]
It may be a good idea to split this up into a number of PRs, over a longer period of time, so it's easier to review.

**Comment by JLockerman** [comment]
(update scripts will take a little more work to update, since code referencing the old version needs to use the old name, while that referencing the new version needs to use the new, if anyone has a better way to do this than by manual inspection please let me know)

File: src/dimension.c [hunk 1]
```diff
@@ -988,18 +988,18 @@ dimension_validate_info(DimensionInfo *info)
 }
 
 void
-dimension_add_from_info(DimensionInfo *info)
+ts_dimension_add_from_info(DimensionInfo *info)
```
Comment (erimatnor) [inline src/dimension.c]: This function is not exported so need not have the `ts_` prefix

File: src/dimension.c [hunk 2]
```diff
@@ -756,7 +756,7 @@ dimension_interval_to_internal_test(PG_FUNCTION_ARGS)
 }
 
 static void
-dimension_add_not_null_on_column(Oid table_relid, char *colname)
+ts_dimension_add_not_null_on_column(Oid table_relid, char *colname)
```
Comment (erimatnor) [inline src/dimension.c]: this is a static and non-exported function so it need not have the `ts_` prefix

File: src/hypertable.c [hunk 3]
```diff
@@ -948,7 +948,7 @@ hypertable_has_tuples(Oid table_relid, LOCKMODE lockmode)
 }
 
 static void
-hypertable_create_schema(const char *schema_name)
+ts_hypertable_create_schema(const char *schema_name)
```
Comment (erimatnor) [inline src/hypertable.c]: This one doesn't need the `ts_` prefix.

File: src/tablespace.c [hunk 4]
```diff
@@ -466,13 +466,13 @@ tablespace_attach(PG_FUNCTION_ARGS)
...
 
-	tablespace_attach_internal(tspcname, hypertable_oid, if_not_attached);
+	ts_tablespace_attach_internal(tspcname, hypertable_oid, if_not_attached);
 
 	PG_RETURN_VOID();
 }
 
 void
-tablespace_attach_internal(Name tspcname, Oid hypertable_oid, bool if_not_attached)
+ts_tablespace_attach_internal(Name tspcname, Oid hypertable_oid, bool if_not_attached)
```
Comment (erimatnor) [inline src/tablespace.c]: Doesn't need `ts_` prefix.

File: src/tablespace.c [hunk 5]
```diff
@@ -544,7 +544,7 @@ tablespace_attach_internal(Name tspcname, Oid hypertable_oid, bool if_not_attach
 }
 
 static int
-tablespace_detach_one(Oid hypertable_oid, const char *tspcname, Oid tspcoid, bool if_attached)
+ts_tablespace_detach_one(Oid hypertable_oid, const char *tspcname, Oid tspcoid, bool if_attached)
```
Comment (erimatnor) [inline src/tablespace.c]: Doesn't need `ts_` prefix.

File: src/tablespace.c [hunk 6]
```diff
@@ -580,7 +580,7 @@ tablespace_detach_one(Oid hypertable_oid, const char *tspcname, Oid tspcoid, boo
 }
 
 static int
-tablespace_detach_all(Oid hypertable_oid)
+ts_tablespace_detach_all(Oid hypertable_oid)
```
Comment (erimatnor) [inline src/tablespace.c]: Doesn't need `ts_` prefix.

File: test/src/net/test_conn.c [hunk 7]
```diff
@@ -9,10 +9,10 @@
 
 #define MAX_RESULT_SIZE	2048
 
-TS_FUNCTION_INFO_V1(test_conn);
+TS_FUNCTION_INFO_V1(ts_test_conn);
 
 Datum
-test_conn(PG_FUNCTION_ARGS)
+ts_test_conn(PG_FUNCTION_ARGS)
```
Comment (erimatnor) [inline test/src/net/test_conn.c]: Strictly, our test code doesn't need the `ts_` prefix, but I guess it doesn't hurt for consistency.

**Comment by JLockerman** [comment]
Fixed extra renames.

File: test/src/net/test_conn.c (hunk 7 above)
Comment (JLockerman) [reply test/src/net/test_conn.c]: That's what I figured.lso @cevian is talking

**Pull request merged**